### PR TITLE
Add cloud pointer to neg linker and backend syncer

### PR DESCRIPTION
**Pull request by spencerhance**
Remove BackendPool downcast by adding a cloud pointer to the neg linker and backend syncer

### CODE REVIEW

**Comment by rramkumar1** [comment]
/ok-to-test

@spencerhance looks like this needs a rebase

**Comment by spencerhance** [comment]
@rramkumar1 thanks, just fixed it now

File: pkg/backends/syncer.go [hunk 1]
```diff
@@ -43,11 +44,13 @@ var _ Syncer = (*backendSyncer)(nil)
 func NewBackendSyncer(
 	backendPool Pool,
```
Comment (rramkumar1) [inline pkg/backends/syncer.go]: Maybe I'm missing something but doesn't seem like backendPool is used anymore? Can we remove it if that is the case?

File: pkg/backends/neg_linker.go [hunk 2]
```diff
@@ -34,11 +36,13 @@ var _ Linker = (*negLinker)(nil)
 func NewNEGLinker(
 	backendPool Pool,
```
Comment (rramkumar1) [inline pkg/backends/neg_linker.go]: Same comment as below regarding backendPool not being used any longer.

File: pkg/backends/syncer.go (hunk 1 above)
Comment (spencerhance) [reply pkg/backends/syncer.go]: We definitely still use it in the syncer

File: pkg/backends/neg_linker.go (hunk 2 above)
Comment (spencerhance) [reply pkg/backends/neg_linker.go]: We only use it in the neg linker test

File: pkg/backends/syncer.go [hunk 3]
```diff
@@ -43,11 +44,13 @@ var _ Syncer = (*backendSyncer)(nil)
...
 	healthChecker healthchecks.HealthChecker,
-	namer *utils.Namer) Syncer {
+	namer *utils.Namer,
+	cloud *gce.Cloud) Syncer {
```
Comment (bowei) [inline pkg/backends/syncer.go]: [go style] we don't break line for args like this

**Pull request merged**